modules/kommune/model/kommune.class.php

```diff
@@ -0,0 +1,180 @@
+<?
+/*
+ * Kommune class
+ *
+ */
+
+class kommune {
+
+  private $db_table = 'kommune';
+  private $all_kommunes = array();
+  private $kommunes_in_db_by_id = array();
+  private $kommunes_in_db_by_number = array();
+  private $kommune_data_for_select = array();
+
+  // Table columns
+  public $KommuneID         = NULL;
+  public $KommuneNumber     = NULL;
+  public $KommuneName       = NULL;
+  public $County            = NULL;
+  public $Sone              = NULL;
+  public $BankAccountNumber = NULL;
+  public $OrgNumber         = NULL;
+  public $OrgName           = NULL;
+  public $OrganisationForm  = NULL;
+  public $Comments          = NULL;
+
+  // Linked to the AGA table by Sone code
+  public $TaxPercent        = NULL;
+
+  public function __construct($id = NULL) {
+    self::init();
+    // load the kommune if the id given
+    if (!is_null($id)) self::load($id);
+  }
+
+  // Initialize, load all kommunes from csv file, all the kommunes from the database
+  // and all the tax percentage per zone code
+  public function init() {
+    global $_SETUP, $_lib;
+    $this->all_kommunes = array();
+    $this->kommunes_in_db_by_id = array();
+    $this->kommunes_in_db_by_number = array();
+
+    $csv_file_url = $_SETUP['HOME_DIR'] . $_SETUP['KOMMUNE_CSV'];
+    if (($kommune_cvs_file = fopen($csv_file_url, "r")) !== FALSE) {
+      while (($csvdata = fgetcsv($kommune_cvs_file)) !== FALSE) {
+        $kommune_object = self::kommune_object_from_csvdata($csvdata);
+        $this->all_kommunes[$kommune_object->KommuneNumber] = $kommune_object;
+        $this->kommune_data_for_select[$kommune_object->KommuneNumber] = implode($csvdata, "; ");
+      }
+      fclose($kommune_cvs_file);
+    }
+
+    // get all tax precentage by zone code
```

Comment:
move above to have this data in function call for `kommune_object_from_csvdata()` on line 47